addons/project/models/project.py

```diff
@@ -750,19 +718,20 @@ def update_date_end(self, stage_id):
     # Subtasks
     # ---------------------------------------------------
 
-    def _subtask_default_fields(self):
-        """ Return the list of field name for default value when creating a subtask """
-        return ['partner_id', 'email_from']
+    @api.depends('parent_id.partner_id', 'project_id.partner_id')
+    def _compute_partner_id(self):
+        for task in self:
+            task.partner_id = task.partner_id or task.parent_id.partner_id or task.project_id.partner_id
+
+    @api.depends('partner_id.email', 'parent_id.email_from')
+    def _compute_email_from(self):
+        for task in self:
+            task.email_from = task.partner_id.email or task.email_from or task.parent_id.email_from
 
-    def _subtask_values_from_parent(self, parent_id):
-        """ Get values for substask implied field of the given"""
-        result = {}
-        parent_task = self.env['project.task'].browse(parent_id)
-        for field_name in self._subtask_default_fields():
-            result[field_name] = parent_task[field_name]
-        # special case for the subtask default project
-        result['project_id'] = parent_task.project_id.subtask_project_id
-        return self._convert_to_write(result)
+    @api.depends('parent_id.project_id.subtask_project_id')
+    def _compute_project_id(self):
+        for task in self:
+            task.project_id = task.project_id or task.parent_id.project_id.subtask_project_id
```

Comment:
ping @Julien00859 
```python
task.project_id = task.project_id or ...
```
?

Reply:
en général on ne réécris pas sur le field (la réécriture est effective), je préfère `if not task.project_id: task.project_id = task.parent_id.project_id.subtask_project_id`

Reply:
ok merci :)